### PR TITLE
Ensure GitHub username is lower-case more aggressively

### PR DESCRIPTION
**Pull request by meatballhat**

### CODE REVIEW

File: fuzzbucket_client/__main__.py [hunk 1]
```diff
@@ -501,6 +518,7 @@ def main(sysargs: typing.List[str] = sys.argv[:]) -> int:
     parser_delete_key.set_defaults(func=client.delete_key)
 
     known_args, unknown_args = parser.parse_known_args(sysargs[1:])
+    known_args = _normalize_known_args(known_args)
```
Comment (mmarchetti) [inline fuzzbucket_client/__main__.py:521]: You might consider using a custom `type` for the `user` argument, as described [here](https://docs.python.org/3/library/argparse.html#type). That would avoid the need to clone the namespace to update the username.

File: fuzzbucket/app.py [hunk 2]
```diff
@@ -74,6 +74,14 @@ def set_user():
         if user is None:
             user, source = request.args.get("user"), "qs"
         session["user"] = user
+    if (
+        session.get("user") is not None
+        and session["user"] != str(session["user"]).lower()
+    ):
+        session["user"] = str(session["user"]).lower()
```
Comment (mmarchetti) [inline fuzzbucket/app.py:81]: The two instances of this line could be deduplicated with something like
```
    if session.get("user") is not None:
        lower_user = str(session["user"]).lower()
        if session["user"] != lower_user:
            session["user"] = lower_user
            log.debug(...)
```

**Review by mmarchetti — APPROVED** [review]

File: fuzzbucket_client/__main__.py (hunk 1 above)
Comment (meatballhat) [reply fuzzbucket_client/__main__.py:521]: I had a version of this where I'd used `type` 😂  but having the logging side effect in there felt weird. WDYT?

Comment (mmarchetti) [reply fuzzbucket_client/__main__.py:521]: I'm OK either way

**Pull request merged**